.github/workflows/docker-builder-publisher.yaml

```diff
@@ -11,6 +11,10 @@ jobs:
   build-and-publish:
     runs-on: ubuntu-latest
     environment: CD
```

Comment:
federation may requite environment to be called `main`

Reply:
You may need to create new action that build image independently into our internal registries that has `main` as an environment